### PR TITLE
Add Voxelarea unittests

### PR DESCRIPTION
**Pull request by nerzhul**

### CODE REVIEW

File: src/unittest/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,7 @@ set (UNITTEST_SRCS
 	${CMAKE_CURRENT_SOURCE_DIR}/test_socket.cpp
 	${CMAKE_CURRENT_SOURCE_DIR}/test_threading.cpp
 	${CMAKE_CURRENT_SOURCE_DIR}/test_utilities.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/test_voxelarea.cpp
```
Comment (SmallJoker) [inline src/unittest/CMakeLists.txt]: Missing android makefile entry.

File: src/unittest/test_voxelarea.cpp [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+void TestVoxelArea::test_volume()
+{
+	VoxelArea v1(v3s16(-1337, 447, -789), v3s16(-147, -9547, 669));
+	UASSERTEQ(s32, v1.getVolume(), -184657133);
+}
+
+void TestVoxelArea::test_contains_voxelarea()
+{
+	VoxelArea v1(v3s16(-1337, -9547, -789), v3s16(-147, 750, 669));
+	UASSERTEQ(bool, v1.contains(VoxelArea(v3s16(-200, 10, 10), v3s16(-150, 10, 10))),
```
Comment (SmallJoker) [inline src/unittest/test_voxelarea.cpp]: Could you please test areas that border to another to check for offset errors?

**Comment by nerzhul** [comment]
@SmallJoker do you have an example ?For Android, we don't build unittests

**Comment by SmallJoker** [comment]
@nerzhul Oh I see. Possibly something like this for the Z-axis border inclusive check:
```C++
VoxelArea v1(v3s16(-1337, -9547, -789), v3s16(-147, 750, 669));
UASSERTEQ(bool, v1.contains(VoxelArea(v3s16(-800, -400, 669), v3s16(-500, 200, 669))),
		true);
UASSERTEQ(bool, v1.contains(VoxelArea(v3s16(-800, -400, 670), v3s16(-500, 200, 670))),
		false);
```

**Pull request merged**